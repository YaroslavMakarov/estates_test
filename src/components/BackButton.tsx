import React from 'react';
import { Container, Button } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';
import { useHistory } from 'react-router-dom';

const BackButton = () => {
    const history = useHistory();

    return (
        <Container
            fluid
            className="p-0"
        >
            <Button
                variant="dark"
                className="mb-3"
                onClick={() => history.goBack()}
            >
                Back
            </Button>
        </Container>
    );
};

export default BackButton;