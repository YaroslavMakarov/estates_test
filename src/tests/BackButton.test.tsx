import React from 'react';
import { render } from '@testing-library/react';

import BackButton from '../components/BackButton';

describe('BackButton component', () => {
    const wrapper = render(<BackButton />);

    it('get BackButton snapshot', () => {
        expect(wrapper.baseElement).toMatchSnapshot();
    });
});