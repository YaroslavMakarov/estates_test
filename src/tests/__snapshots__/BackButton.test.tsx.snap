// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<body>
  <div>
    <div
      class="p-0 container-fluid"
    >
      <button
        class="mb-3 btn btn-dark"
        type="button"
      >
        Back
      </button>
    </div>
  </div>
</body>
`;

exports[`BackButton component get BackButton snapshot 1`] = `
<body>
  <div>
    <div
      class="p-0 container-fluid"
    >
      <button
        class="mb-3 btn btn-dark"
        type="button"
      >
        Back
      </button>
    </div>
  </div>
</body>
`;
